src/private_broadcast.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2023-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://opensource.org/license/mit/.
+
+#include <private_broadcast.h>
+#include <util/check.h>
+
+/// If a transaction is not received back from the network for this duration
+/// after it is broadcast, then we  consider it stale / for rebroadcasting.
+static constexpr auto STALE_DURATION{1min};
+
+bool PrivateBroadcast::Add(const CTransactionRef& tx) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    const Txid& txid = tx->GetHash();
+    LOCK(m_mutex);
+    auto [pos, inserted] = m_by_txid.emplace(txid, TxWithPriority{.tx = tx, .priority = Priority{}});
+    if (inserted) {
+        m_by_priority.emplace(Priority{}, txid);
+    }
+    return inserted;
+}
+
+std::optional<size_t> PrivateBroadcast::Remove(const CTransactionRef& tx) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    LOCK(m_mutex);
+    auto iters = Find(tx->GetHash());
+    if (!iters || iters->by_txid->second.tx->GetWitnessHash() != tx->GetWitnessHash()) {
+        return std::nullopt;
+    }
+    const size_t num_broadcasted{iters->by_priority->first.num_broadcasted};
+    m_by_priority.erase(iters->by_priority);
+    m_by_txid.erase(iters->by_txid);
+    return num_broadcasted;
```

Comment:
The `extract` keeps giving:
```suggestion
    if (auto iters{Find(tx->GetHash())}; iters && iters->by_txid->second.tx->GetWitnessHash() == tx->GetWitnessHash()) {
        m_by_txid.erase(iters->by_txid);
        return m_by_priority.extract(iters->by_priority).key().num_broadcasted;
    }
    return std::nullopt;
```

Reply:
> Reverted the logic

👍 

> but kept the num_broadcasted variable and the two erases, it is more debugger friendly.

Isn't this the exact scenario for why `extract` was introduced in C++20? Or was I mis-suggesting it?